test/lit_tests/communication/extend.mlir

```diff
@@ -0,0 +1,28 @@
+// RUN: enzymexlamlir-opt --optimize-communication %s | FileCheck %s
+
+sdy.mesh @mesh1 = <["z"=1, "x"=4, "y"=4]>
+func.func @main(%arg0: tensor<20x24x80xf64> {sdy.sharding = #sdy.sharding<@mesh1, [{"z"}, {"y"}, {"x"}]>}) -> (tensor<1x10x82xf64> {sdy.sharding = #sdy.sharding<@mesh1, [{"z"}, {"y"}, {"x"}]>}) {
+    %0 = stablehlo.slice %arg0 [11:12, 7:17, 0:80] {sdy.sharding = #sdy.sharding_per_value<[<@mesh1, [{"z"}, {"y"}, {"x"}]>]>} : (tensor<20x24x80xf64>) -> tensor<1x10x80xf64>
+    %1 = "enzymexla.extend"(%0) <{dimension = 2 : i64, lhs = 1 : i64, rhs = 1 : i64}> {sdy.sharding = #sdy.sharding_per_value<[<@mesh1, [{"z"}, {"y"}, {"x"}]>]>} : (tensor<1x10x80xf64>) -> tensor<1x10x82xf64>
+    return %1 : tensor<1x10x82xf64>
+}
+
+// CHECK-LABEL:   func.func @main
+// CHECK: sdy.manual_computation
+// CHECK: "stablehlo.if"
+// CHECK: "stablehlo.if"
```

Comment:
can you actually fill out the entire test here? (you can be lazy and copy the output, prepend // CHECK-NEXT: in front of all lines except the first